conans/client/installer.py

```diff
@@ -107,6 +107,10 @@ def _prepare_sources(self, conanfile, pref, package_layout, conanfile_path, sour
     def _build(self, conanfile, pref, build_folder):
         # Read generators from conanfile and generate the needed files
         logger.info("GENERATORS: Writing generators")
+
+        if conanfile.lyt:
+            build_folder = os.path.join(build_folder, conanfile.lyt.build_install_folder)
+
         write_generators(conanfile, build_folder, self._output)
```

Comment:
FIXME: This method is wrong/incomplete, probably the build_folder should be modified only for the write_generators.